src/parser/parser.cpp

```diff
@@ -661,14 +661,14 @@ Command* Parser::nextCommand()
   return cmd;
 }
 
-Expr Parser::nextExpression()
+api::Term Parser::nextExpression()
 {
   Debug("parser") << "nextExpression()" << std::endl;
   d_resourceManager->spendResource(ResourceManager::Resource::ParseStep);
   Expr result;
   if (!done()) {
     try {
-      result = d_input->parseExpr();
+      result = d_input->parseExpr().getExpr();
```

Comment:
Why not just make the `result` a `Term`?

Reply:
It is on my branch, this change touches fewer lines (will have fewer merge conflicts this way).

Reply:
Actually i guess its even 2 vs 2 haha. Anyways that was the reason.

Reply:
Actually nevermind, this way has fewer merge conflicts :)  Anyways that was supposely the reason.